src/compiler/tsbuildPublic.ts

```diff
@@ -1578,16 +1744,36 @@ namespace ts {
 
         // Up to date
         return {
-            type: pseudoUpToDate ? UpToDateStatusType.UpToDateWithUpstreamTypes : UpToDateStatusType.UpToDate,
+            type: pseudoUpToDate ?
+                UpToDateStatusType.UpToDateWithUpstreamTypes :
+                pseudoInputUptodate ?

```

Comment:
Please consistently use "UpToDate", rather than other capitalizations.

As discussed offline, this would probably be clearer as "inputPseudoUpToDate".